<?xml version="1.0"?>
<launch>

    <arg name="path2cam_calib_yaml" default="$(find pp_alvium_driver)/calib/220314_nir_2point5k.yaml" doc="relative path to the camera calibration yaml file"/>
    <arg name="camera_id" default="DEV_1AB22C00C28B" doc="ID of the camera, e.g. DEV_1AB22C00A470"/>
    <arg name="exposure_time" default="2200" doc="Absolute value in microseconds (10^-6s) (no max or something)"/>
    <arg name="hardware_delay_correction" default="0.0125" doc="Absolute value in seconds. This is due to the lag in transfering imgs from the cam to the callback"/>
    <arg name="published_img_topic" default="/pp_nir/raw" doc="name of the camera images topic"/>
    <arg name="publish_camera_info" default="true" doc="Write true to publish camera info too"/>
    <arg name="published_caminfo_topic" default="/pp_nir/camera_info" doc="name of the camera info topic"/>


<!--     <remap from="img_topic_subscribed" to="cam0/image_raw" /> -->
<!--     <remap from="pattern_topic_published" to="cam0/calib_pattern_recognised" /> -->
    <node pkg="pp_alvium_driver" name="ros_alvium_nir" type="ros_alvium_nir.py" output="screen">
            <param name="path2cam_calib_yaml" value="$(arg path2cam_calib_yaml)"/>
            <param name="camera_id" value="$(arg camera_id)"/>
            <param name="exposure_time" value="$(arg exposure_time)"/>
            <param name="hardware_delay_correction" value="$(arg hardware_delay_correction)"/>
            <param name="published_img_topic" value="$(arg published_img_topic)"/>
            <param name="publish_camera_info" value="$(arg publish_camera_info)"/>
            <param name="published_caminfo_topic" value="$(arg published_caminfo_topic)"/>
            <param name="frequency" value="60"/>
<!--         <rosparam file="$(find pp_alvium_driver)/config/params_nir.yaml" command="load"/> -->
<!--         <remap from="pp/rgb_raw" to="pp/nir_raw"/> -->
    </node>

</launch>
